API.md

```diff
@@ -1,445 +1,648 @@
+```
+
+
+
+|Bucket operations | Object operations| Presigned operations |
+|---|---|---|
+| [`make_bucket`](#make_bucket)  | [`get_object`](#get_object)  | [`presigned_get_object`](#presigned_get_object)  |
+|[`list_buckets`](#list_buckets)   | [`put_object`](#put_object)  | [`presigned_put_object`](#presigned_put_object)  |
+| [`bucket_exists`](#bucket_exists)  |[`stat_object`](#stat_object)   |[`presigned_post_policy`](#presigned_post_policy)   |
+|[`remove_bucket`](#remove_bucket)   | [`remove_object`](#remove_object)  |   |
+| [`list_objects`](#list_objects)  | [`remove_incomplete_upload`](#remove_incomplete_upload)  |   | 
+|[`list_incomplete_uploads`](#list_incomplete_uploads)   |  [`fput_object`](#fput_object) |   |
+|  |[`fget_object`](#fget_object)  |  | 
+|  |[`get_partial_object`](#get_partial_object)  |  | 
+
+## 1. Constructor
+---------------------------------------
+<a name="Minio">
+#### Minio(endpoint, access_key=None, secret_key=None, secure=True)
+
+|   |
+|---|
+| `Minio(endpoint, access_key=None, secret_key=None, secure=True)`  |
+| Initializes a new client object.  |
+
+__Parameters__
+
+
+| Param  |  Type | Description  |
+|---|---|---|
+| `endpoint`  | _string_  | S3 object storage endpoint.  |
+| `access_key`  | _string_  | Access key for the object storage endpoint. (Optional if you need anonymous access).  |
+|  `secret_key` | _string_  |  Secret key for the object storage endpoint. (Optional if you need anonymous access). |
+| `secure`  |_bool_   | Set this value to `True` to enable secure (HTTPS) access. (Optional defaults to `True`).  |
+
+__Example__
+
+``1. Minio``
+```py
+from minio import Minio
+from minio.error import ResponseError
+
+minioClient = Minio('play.minio.io:9000',
+                    access_key='Q3AM3UQ867SPQQA43P2F',
+                    secret_key='zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG')
+```
+
+``2.AWS S3``
+```py
+from minio import Minio
+from minio.error import ResponseError
+
+s3Client = Minio('s3.amazonaws.com',
+                 access_key='ACCESS_KEY',
+                 secret_key='SECRET_KEY')
+```
+
+## 2. Bucket operations
+---------------------------------------
+<a name="make_bucket">
+#### make_bucket(bucket_name, location='us-east-1')
+Creates a new bucket.
+
+__Parameters__
+
+<table>
+    <thead>
+        <tr>
+            <th>Param</th>
+            <th>Type</th>
+            <th>Description</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>
+           bucket_name
+            </td>
+            <td> string</td>
+            <td> Name of the bucket.</td>
+            </tr>
+            <tr>
+            <td>location</td>
+            <td>string</td>
+            <td>Default value is us-east-1<br/>
+Location valid values are us-west-1, us-west-2, eu-west-1, eu-central-1, ap-southeast-1, ap-northeast-1, ap-southeast-2, sa-east-1(defaults to us-east-1, optional).</td>
+            </tr>
+               </tbody>
+</table>
+

```

Comment:
Type needs to be italics. if it is not possible here then bring other types to same formatting.